datagen/utils/finality.go

```diff
@@ -49,8 +48,6 @@ func GenRandomPubRandList(r *rand.Rand, numPubRand uint64) (*datagen.RandListInf
 }
 
 func GenCommitPubRandListMsg(startHeight uint64, numPubRand uint64, pubRandIndex uint64, dir string) (*datagen.RandListInfo, *btcec.PrivateKey) {
-	r := rand.New(rand.NewSource(time.Now().UnixNano()))
-
```

Comment:
Yes. This seems to be a good practice. The issue I was facing is as follows. I was using two fresh `r := rand.New(rand.NewSource(time.Now().UnixNano()))` to generate secret keys for covenant members and finality providers in test data. Since the code was fast, the two `r` end up using the same timestamp and being the same value. Then the 1st covenant secret key is same as the 1st finality provider secret key. As Babylon does not accept duplicate keys in the staking script (i.e., any duplication among FP keys, covenant keys and staker key), the tests panicked. Using the same rand source solves this issue.

Reply:
Makes sense overall, but I find difficult to believe that both `r` have the same seed, as it's a timestamp in nanoseconds resolution. Are these routines being called in parallel or so?